tests/models/emformer/test_modeling_emformer.py

```diff
@@ -0,0 +1,405 @@
+
+        # pad input
+        for i in range(len(input_lengths)):
+            input_features[i, input_lengths[i] :] = 0.0
+            attention_mask[i, input_lengths[i] :] = 0
+
+        loss = model(input_features, attention_mask=attention_mask, labels=labels).loss.item()
+
+        self.parent.assertTrue(isinstance(loss, float))
+
+    def check_labels_out_of_vocab(self, config, input_features, *args):
+        model = EmformerForRNNT(config)
+        model.to(torch_device)
+        model.train()
+
+        input_features = input_features[:3]
+
+        input_lengths = [input_features.shape[-1] // i for i in [4, 2, 1]]
+        max_length_labels = model._get_time_reduced_output_lengths(torch.tensor(input_lengths))
+        labels = ids_tensor((input_features.shape[0], max(max_length_labels) - 2), model.config.vocab_size + 100)
+
+        with self.parent.assertRaises(ValueError):
+            model(input_features, labels=labels)
+
+    def prepare_config_and_inputs_for_common(self):
+        config, input_features, attention_mask = self.prepare_config_and_inputs()
+        inputs_dict = {"input_features": input_features, "attention_mask": attention_mask}
+        return config, inputs_dict
+
+
+@require_torch
+class EmformerModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (EmformerModel, EmformerForRNNT) if is_torch_available() else ()
+    test_pruning = False
+    test_headmasking = False
+    test_torchscript = False
+
+    def setUp(self):
+        self.model_tester = EmformerModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=EmformerConfig, hidden_size=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_labels_out_of_vocab(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.check_labels_out_of_vocab(*config_and_inputs)
+
+    # Emformer has no inputs_embeds
+    def test_inputs_embeds(self):
+        pass
+
+    # `input_ids` is renamed to `input_features`
+    def test_forward_signature(self):
+        pass
+
+    # Emformer cannot resize token embeddings
+    # since it has no tokens embeddings
+    def test_resize_tokens_embeddings(self):
+        pass
+
+    # Emformer has no inputs_embeds
+    # and thus the `get_input_embeddings` fn
+    # is not implemented
+    def test_model_common_attributes(self):
+        pass
+
+    @is_pt_flax_cross_test
+    # not implemented in Flax
+    def test_equivalence_flax_to_pt(self):
+        pass
+
+    @is_pt_flax_cross_test
+    # not implemented in Flax
+    def test_equivalence_pt_to_flax(self):
+        pass
+
+    def test_retain_grad_hidden_states_attentions(self):
```

Comment:
Let's remove / overwrite the test - it's a nice to have but absolutely not necessary